crates/core/src/client/messages.rs

```diff
@@ -22,7 +24,38 @@ pub trait ServerMessage: Sized {
     fn serialize(self, protocol: Protocol) -> DataMessage {
         match protocol {
             Protocol::Text => self.serialize_text().to_json().into(),
-            Protocol::Binary => self.serialize_binary().encode_to_vec().into(),
+            Protocol::Binary => {
+                let msg_bytes = self.serialize_binary().encode_to_vec();
+                let reader = &mut &msg_bytes[..];
+
+                // TODO(perf): Compression should depend on message size and type.
+                //
+                // SubscriptionUpdate messages will typically be quite large,
+                // while TransactionUpdate messages will typically be quite small.
+                //
+                // If we are optimizing for SubscriptionUpdates,
+                // we want a large buffer.
+                // But if we are optimizing for TransactionUpdates,
+                // we probably want to skip compression altogether.
+                //
+                // For now we choose a reasonable middle ground,
+                // which is to compress everything using a 32KB buffer.
+                const BUFFER_SIZE: usize = 32 * 1024;
+                // Again we are optimizing for compression speed,
+                // so we choose the lowest (fastest) level of compression.
```

Comment:
@joshua-spacetime may know this, but I do not.

Reply:
In general we can't say, since it depends on many factors. For an internal workload we see a 10X compression ratio for subscription update messages, but again that's very workload specific and probably doesn't generalize.